sbt-version-policy/src/main/scala/sbtversionpolicy/withsbtrelease/ReleaseVersion.scala

```diff
@@ -0,0 +1,123 @@
+package sbtversionpolicy.withsbtrelease
+
+import sbtversionpolicy.Compatibility
+import sbtversionpolicy.SbtVersionPolicyPlugin.aggregatedAssessedCompatibilityWithLatestRelease
+import sbtversionpolicy.SbtVersionPolicyPlugin.autoImport.versionPolicyAssessCompatibility
+import sbt.*
+import sbtversionpolicy.SbtVersionPolicyMima.autoImport.versionPolicyPreviousVersions
+
+/** Convenient methods to integrate with the plugin sbt-release */
+object ReleaseVersion {
+
+  private val qualifierVariableName = "VERSION_POLICY_RELEASE_QUALIFIER"
+
+  /**
+   * @return a [release version function](https://github.com/sbt/sbt-release?tab=readme-ov-file#custom-versioning)
+   *         that bumps the patch, minor, or major version number depending on the provided
+   *         compatibility level.
+   * @param qualifier Optional qualifier to append to the version (e.g. `"-RC1"`). By default, it tries to read
+   *                  it from the environment variable VERSION_POLICY_RELEASE_QUALIFIER.
+   */
+  def fromCompatibility(compatibility: Compatibility, qualifier: String = sys.env.getOrElse(qualifierVariableName, "")): String => String = {
+    val maybeBump =
+      compatibility match {
+        case Compatibility.None => Some(Version.Bump.Major)
+        case Compatibility.BinaryCompatible => Some(Version.Bump.Minor)
+        case Compatibility.BinaryAndSourceCompatible => None // No need to bump the patch version, because it has already been bumped when sbt-release set the next release version
+      }
+    { (currentVersion: String) =>
+      val versionWithoutQualifier =
+        Version(currentVersion)
+          .getOrElse(Version.formatError(currentVersion))
+          .withoutQualifier
+      val bumpedVersion =
+        (maybeBump match {
+          case Some(bump) => versionWithoutQualifier.bump(bump)
+          case None => versionWithoutQualifier
+        }).string
+      bumpedVersion + qualifier
+    }
+  }
+
+  private def fromAssessedCompatibility(qualifier: String)(assessCompatibility: Def.Initialize[Task[Compatibility]]): Def.Initialize[Task[String => String]] =
+    Def.ifS(Def.task {
+      versionPolicyPreviousVersions.value.isEmpty
+    })(Def.task {
+      // If there are no previous versions to assess the compatibility with,
+      // fallback to the default release version function, which drops the qualifier
+      // from the version set in the file `version.sbt`
+      // (e.g., "1.0.0-SNAPSHOT" => "1.0.0")
+      (version: String) =>
+        Version(version)
+          .map(_.withoutQualifier.string + qualifier)
+          .getOrElse(Version.formatError(version))
```

Comment:
Ah clever, I like it! In order to work out the new version number if there's no previous released version to compare compatibility with, we just read the current snapshot version specified in `version.sbt` (eg "1.0.0-SNAPSHOT") and use it without the qualifier (eg "1.0.0") 👍